docs/guide/dep-pre-bundling.md

```diff
@@ -28,7 +28,9 @@ This is Vite performing what we call "dependency pre-bundling". This process ser
 
    By pre-bundling `lodash-es` into a single module, we now only need one HTTP request instead!
 
-Note that this only applies in development mode.
+::: tip NOTE
+Dependency pre-bundling only applies in development mode, and uses `esbuild` to convert dependencies to ESM. In production builds, `@rollup/plugin-commonjs` is used instead. This is safe in most cases, but may cause small inconsistencies for some dependencies.
```

Comment:
I think the last line isn't needed. We could list the known inconsistencies or leave it only showing what is used in dev and build. 
```suggestion
Dependency pre-bundling only applies in development mode, and uses `esbuild` to convert dependencies to ESM. In production builds, `@rollup/plugin-commonjs` is used instead.
```